### PR TITLE
Rename "inverted index" to "full-text index"

### PR DESCRIPTION
**Pull request by rschu1ze**
This PR renames "inverted indexes" to "full-text indexes", see https://github.com/ClickHouse/ClickHouse/issues/50268#issuecomment-1565101229

*Review note*: Due to conflicts with "full-text indexes" as existing internal name for bloom filter indexes, this PR consists of three steps:
- https://github.com/ClickHouse/ClickHouse/commit/f0faac2e8bcef83289e858e4cbfe374f68cf2ce4 moves everything related to bloom filter indexes into a single header / source file. This is technically unrelated to the original purpose, but makes the code organization consistent with the other secondary index implementations.
- https://github.com/ClickHouse/ClickHouse/commit/0ae422d33c0538f759c64444cc8b7a6f8f2a5b63 renames `MergeTreeIndexFullText*` to `MergeTreeIndexBloomFilterText*`.
- https://github.com/ClickHouse/ClickHouse/pull/62884/commits/56e32e0f996d8a0838ef9f89f7a1db0bc5b14a38 renames `MergeTreeIndexInverted*` to `MergeTreeIndexFullText*`.

### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Renamed "inverted indexes" to "full-text indexes" which is a less technical / more user-friendly name. This also changes internal table metadata and breaks tables with existing (experimental) inverted indexes. Please make to drop such indexes before upgrade and re-create them after upgrade.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 3266823d5f488a2aa21c413fba8ad40444811e50 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62884/3266823d5f488a2aa21c413fba8ad40444811e50/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
How about `full_text_ngram`, `full_text_token`?

**Comment by rschu1ze** [comment]
> How about `full_text_ngram`, `full_text_token`?

You surely mean
- `full_text_token` as alias for `full_text(0)`, and
- `full_text_ngram` as alias for `full_text(N)`

during index creation. I generally like this idea and it seems (kind of) consistent with the naming of bloom filter indexes `tokenbf_v1` and `ngrambf_v1`. But perhaps we like to specify tokenizers in future in more detail (think of stemming, lemmatization, etc.). So instead of hard-coding some index type names now, we should first check which flexibility we really need. I better do that in another PR.

File: src/Storages/MergeTree/MergeTreeDataPartWriterOnDisk.cpp [hunk 1]
```diff
@@ -471,7 +471,7 @@ void MergeTreeDataPartWriterOnDisk::fillSkipIndicesChecksums(MergeTreeData::Data
         /// Register additional files written only by the inverted index. Required because otherwise DROP TABLE complains about unknown
         /// files. Note that the provided actual checksums are bogus. The problem is that at this point the file writes happened already and
         /// we'd need to re-open + hash the files (fixing this is TODO). For now, CHECK TABLE skips these four files.
-        if (typeid_cast<const MergeTreeIndexInverted *>(&*skip_indices[i]) != nullptr)
+        if (typeid_cast<const MergeTreeIndexFullText *>(&*skip_indices[i]) != nullptr)
```
Comment (antaljanosbenjamin) [inline src/Storages/MergeTree/MergeTreeDataPartWriterOnDisk.cpp:474]: In the docs above maybe replace `inverted` by `full-text`.

File: src/Storages/MergeTree/MutateTask.cpp [hunk 2]
```diff
@@ -653,7 +653,7 @@ static NameSet collectFilesToSkip(
         files_to_skip.insert(index->getFileName() + mrk_extension);
 
         // Skip all inverted index files, for they will be rebuilt
-        if (dynamic_cast<const MergeTreeIndexInverted *>(index.get()))
+        if (dynamic_cast<const MergeTreeIndexFullText *>(index.get()))
```
Comment (antaljanosbenjamin) [inline src/Storages/MergeTree/MutateTask.cpp:656]: Replace `inverted` by `full-text` in the comment.

File: src/Core/Settings.h [hunk 3]
```diff
@@ -233,7 +233,7 @@ class IColumn;
     M(Bool, do_not_merge_across_partitions_select_final, false, "Merge parts only in one partition in select final", 0) \
     M(Bool, split_parts_ranges_into_intersecting_and_non_intersecting_final, true, "Split parts ranges into intersecting and non intersecting during FINAL optimization", 0) \
     M(Bool, split_intersecting_parts_ranges_into_layers_final, true, "Split intersecting parts ranges into layers during FINAL optimization", 0) \
-    M(Bool, allow_experimental_inverted_index, false, "If it is set to true, allow to use experimental inverted index.", 0) \
+    M(Bool, allow_experimental_inverted_index, false, "If it is set to true, allow to use experimental fulltext (inverted) index.", 0) \
```
Comment (antaljanosbenjamin) [inline src/Core/Settings.h:236]: As the feature is experimental and we are changing the declaration of them I think we can also change the settings. If somebody didn't use the feature, they won't have any issues. If somebody uses the feature, then they will have bigger problems with metadata and stuff than a single setting name.

Comment (rschu1ze) [reply src/Core/Settings.h:236]: You are generally right. But: I believe (not 100% sure) that setting `allow_experimental_inverted_index` is force-disabled in ClickHouse cloud via config file. Renaming the setting name will cause unnecessary trouble :-/ At some point, the setting will go away so `allow_experimental_inverted_index` should be good for the time being.

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by Algunenano** [comment]
Changing this to a breaking change. Even though it's experimental, if should appear in the notice list

**Pull request merged**